ruby/ql/lib/codeql/files/FileSystem.qll

```diff
@@ -2,6 +2,7 @@
 
 private import codeql.Locations
 private import codeql.util.FileSystem
+private import codeql.ruby.Diagnostics
```

Comment:
This QL module is language agnostic, try not to import any Ruby stuff. Perhaps these predicates are better placed in [`class RubyFile`](https://github.com/github/codeql/blob/25b3d76f71ae84abe8287814d27d280145ab070c/ruby/ql/lib/codeql/ruby/AST.qll#L139)

Although perhaps we should consider `Diagnostics` to be language neutral. Reporting extraction warnings on non-Ruby files, such as xml, json, yaml would actually make sense.

Reply:
I've moved the class to the same file as `class RubyFile`.  I'm not sure if you were saying you'd like it to be converted to a predicate as well.

> Although perhaps we should consider `Diagnostics` to be language neutral.

I can see some benefits to making `Diagnostics` language neutral and shared.  Way beyond the scope of this PR though.